crowbar_framework/app/models/service_object.rb

```diff
@@ -1145,12 +1143,9 @@ def apply_role(role, inst, in_queue)
               next
             end
 
-            # A new node that we did not know before
-            unless old_nodes.include?(node_name)
-              @logger.debug "add node #{node_name}"
-              pending_node_actions[node_name] = { remove: [], add: [] } if pending_node_actions[node_name].nil?
-              pending_node_actions[node_name][:add] << role_name
-            end
+            pending_node_actions[node_name] ||= { remove: [], add: [] }
+            pending_node_actions[node_name][:add] << role_name
```

Comment:
This is the problematic part. @sayalilunkad , revert this block to the previous state and try with that one.